pkg/apis/v1alpha1/nodeoverlay.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NodeOverlaySpec struct {
+	// Requirements constrain when this NodeOverlay is applied during scheduling simulation.
+	// These requirements can match:
+	// - Well-known labels (e.g., node.kubernetes.io/instance-type, karpenter.sh/nodepool)
+	// - Custom labels from NodePool's spec.template.labels
+	// +kubebuilder:validation:XValidation:message="requirements with operator 'In' must have a value defined",rule="self.all(x, x.operator == 'In' ? x.values.size() != 0 : true)"
+	// +kubebuilder:validation:XValidation:message="requirements operator 'Gt' or 'Lt' must have a single positive integer value",rule="self.all(x, (x.operator == 'Gt' || x.operator == 'Lt') ? (x.values.size() == 1 && int(x.values[0]) >= 0) : true)"
+	// +kubebuilder:validation:MaxItems:=100
+	// +optional
+	Requirements []v1.NodeSelectorRequirement `json:"requirements,omitempty"`
```

Comment:
Should requirements be required? Or does omitting them implicitly mean we should select against all instance types? I'd lean towards requiring it so you need to at least be explicit (e.g. you could do something like this to  select everything)
```yaml
requirements:
- key: node.kubernetes.io/instance-type
  operator: Exists
```

Reply:
Yeah, this makes sense. To be inline with the NodePool requirements, we can make it required